app/safe/internal/state/stats.go

```diff
@@ -28,7 +28,7 @@ var currentState = Status{
 	K8sQueueLen:    0,
 	K8sQueueCap:    0,
 	NumSecrets:     0,
-	lock:           &sync.Mutex{},
+	lock:           &sync.RWMutex{},
```

Comment:
Using a RWMutex is a microoptimization here. Not sure it’s worth it; but also not sure it will cause any harm either. And since it’s just a few characters of change 🤷‍♂️.